### PR TITLE
Task 6 complete before review

### PR DESCRIPTION
**Pull request by lukasz-kukulka**

### CODE REVIEW

File: shm/src/Map.cpp [hunk 1]
```diff
@@ -9,46 +9,37 @@ constexpr size_t COORDINATE_MAX = 99;
...
+    }
+    currentPosition_ = std::make_unique<Island>(vectorOfIslands_[0]);
 }
 
-std::shared_ptr<Island> Map::getIsland(Island::Coordinates& coordinate) {
+std::unique_ptr<Island> Map::getIsland(const Island::Coordinates& coordinate) {
     for (auto const& vec : vectorOfIslands_) {
-        if (vec.getCoordinates() != coordinate) {
-
+        if (vec.getCoordinates() == coordinate) {
```
Comment (mariuszlisowski) [inline shm/src/Map.cpp]: Better name for 'vec' would be 'island' to be clearer what this vector contains.
Use plural of coordinate -> coordinates (as there is positionX & positionY)

File: shm/src/Map.cpp [hunk 2]
```diff
@@ -9,46 +9,37 @@ constexpr size_t COORDINATE_MAX = 99;
...
+            std::unique_ptr<Island> islandPtr = std::make_unique<Island>(vec);
+            return islandPtr;
```
Comment (mariuszlisowski) [inline shm/src/Map.cpp]: Return without creating temporary variable islandPtr:
return std::make_unique<Island>(vec);

Or a raw pointer:
return &vec;

File: shm/src/Map.cpp [hunk 3]
```diff
@@ -9,46 +9,37 @@ constexpr size_t COORDINATE_MAX = 99;
...
+                                               generatedIslandCoords;
+                                   });
+        if (result == vectorOfIslands_.end()) {
+            vectorOfIslands_.push_back(Island(generatedIslandCoords));
+        }
+    }
+    currentPosition_ = std::make_unique<Island>(vectorOfIslands_[0]);
 }
 
-std::shared_ptr<Island> Map::getIsland(Island::Coordinates& coordinate) {
```
Comment (mariuszlisowski) [inline shm/src/Map.cpp:36]: Let's revert to a raw pointer:
Island* Map::getIsland(Island::Coordinates& coordinate)

File: shm/src/Map.cpp [hunk 4]
```diff
@@ -9,46 +9,37 @@ constexpr size_t COORDINATE_MAX = 99;
...
+                                   vectorOfIslands_.cend(),
+                                   [&](const Island& island) {
+                                       return island.getCoordinates() ==
+                                               generatedIslandCoords;
+                                   });
+        if (result == vectorOfIslands_.end()) {
+            vectorOfIslands_.push_back(Island(generatedIslandCoords));
+        }
+    }
+    currentPosition_ = std::make_unique<Island>(vectorOfIslands_[0]);
```
Comment (mariuszlisowski) [inline shm/src/Map.cpp]: Let's revert to a raw pointer:
currentPosition_ = &(vectorOfIslands_[0]);

File: shm/inc/Map.hpp [hunk 5]
```diff
@@ -5,12 +5,12 @@
...
 class Map {
 public:
     Map();
 
-    std::shared_ptr<Island> getIsland(Island::Coordinates&);
+    std::unique_ptr<Island> getIsland(const Island::Coordinates&);
+
 private:
     std::vector<Island> vectorOfIslands_;
     std::unique_ptr<Island> currentPosition_;
```
Comment (mariuszlisowski) [inline shm/inc/Map.hpp]: Let's revert to a raw pointer:
Island* currentPosition_;

File: shm/inc/Map.hpp [hunk 6]
```diff
@@ -5,12 +5,12 @@
 
 #include <memory>
 #include <vector>
-
 class Map {
 public:
     Map();
 
-    std::shared_ptr<Island> getIsland(Island::Coordinates&);
+    std::unique_ptr<Island> getIsland(const Island::Coordinates&);
```
Comment (mariuszlisowski) [inline shm/inc/Map.hpp]: Let's revert to a raw pointer:
Island* getIsland(const Island::Coordinates&);

**Review by mariuszlisowski — CHANGES_REQUESTED** [review]

File: shm/src/Map.cpp [hunk 7]
```diff
@@ -9,46 +9,34 @@ constexpr size_t COORDINATE_MAX = 99;
...
+    }
+    currentPosition_ = &(vectorOfIslands_[0]);
 }
 
-std::shared_ptr<Island> Map::getIsland(Island::Coordinates& coordinate) {
-    for (auto const& vec : vectorOfIslands_) {
-        if (vec.getCoordinates() != coordinate) {
-
+Island* Map::getIsland(const Island::Coordinates& coordinate) {
+    for (auto & island : vectorOfIslands_) {
```
Comment (BarTes8) [inline shm/src/Map.cpp:37]: for (const auto & island : vectorOfIslands_) {

**Review by BarTes8 — APPROVED** [review]

**Pull request merged**